extensions/yaml_support/yaml_support/_parser/base_parser.py

```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env python3
+# Copyright 2004-present Facebook. All Rights Reserved.
+
+from abc import ABCMeta, abstractmethod
+from typing import List, Any
```

Comment:
One more thing LabGraph is going to push is to avoid the use of Any, instead you can use Union or the specific type. In addition, you can consider use PyRe as type check tool, reference: https://pyre-check.org/docs/getting-started/
This is a nice-to-have comment for you to do now.

Reply:
Hello, I was trying to get rid of some of the keywords used however defining the exact type for heterogeneous Dictionaries is a little bit hard and it can lead to really complex types. I'm not sure how can I get around that, could you give me some hints, please?

Reply:
Fyi, the instructions for Windows Subsystem for Linux (WSL) Install can be found in https://pyre-check.org/docs/installation/

You can also define a class for some complex types.

Reply:
Hi, I set up LabGraph on an Ubuntu environment and tried pyre-check, however, Pyre wasn't able to infer some complex types (types that seems recursive)